apps/aecore/lib/aecore/miner/worker.ex

```diff
@@ -290,15 +270,64 @@ defmodule Aecore.Miner.Worker do
   defp filter_transactions_by_fee(txs) do
     miners_fee_bytes_per_token = Application.get_env(:aecore, :tx_data)[:miner_fee_bytes_per_token]
     Enum.filter(txs, fn(tx) ->
-      tx_size_bits = tx
-        |> :erlang.term_to_binary()
-        |> Bits.extract()
-        |> Enum.count()
-      tx_size_bytes = tx_size_bits / 8
+      tx_size_bytes = Pool.get_tx_size_bytes(tx)
       tx.data.fee >= Float.floor(tx_size_bytes / miners_fee_bytes_per_token)
     end)
   end
 
+  defp filter_transactions_by_block_size(txs, current_block_size_bytes, max_block_size_bytes) do
+    first_tx_size_bytes = txs |> Enum.at(0) |> Pool.get_tx_size_bytes()
+
+    filter_transactions_by_block_size(txs, 0, Enum.count(txs), [],
+      current_block_size_bytes, first_tx_size_bytes, max_block_size_bytes)
+  end
+
+  defp filter_transactions_by_block_size(txs, current_tx_index, txs_count, filtered_txs,
+    current_block_size_bytes, next_tx_size_bytes, max_block_size_bytes) do
+
+    current_tx = Enum.at(txs, current_tx_index)
+
+    new_filtered_txs = [current_tx | filtered_txs]
+    next_tx_index = current_tx_index + 1
+    if next_tx_index == txs_count do
+      Enum.reverse(new_filtered_txs)
```

Comment:
why do we use the reverse here and at L300? This might be slow if done for every transaction

Reply:
it is done at the end, upon finishing the filtering

Reply:
ok great!